nexus_constructor/json/shape_reader.py

```diff
@@ -261,9 +264,12 @@ def _get_shape_dataset_from_list(
         :param children: The children list where we expect to find the dataset.
         :return: The dataset if it could be found, otherwise None is returned.
         """
-        for dataset in children:
-            if dataset[CommonKeys.NAME] == dataset_name:
-                return dataset
+        try:
+            for dataset in children:
+                if dataset[NodeType.CONFIG][CommonKeys.NAME] == dataset_name:
+                    return dataset
+        except KeyError:
+            pass  # Create warning message
```

Comment:
except KeyError as error:
       print(error)
maybe?

Reply:
The use of `pass` guarantees that a warning message will be created (if `warning = True`). I have re-written the comment.